### DISCUSSION TITLE
Unsupported wc_ method. wc_sessionPropose

### DISCUSSION THREAD

**Discussion by BigtoLenaDev** [WalletConnect/walletconnect-monorepo#3405] [Developer Support]
**Describe the bug**
Open the modal -> Scan QR code -> The modal shows `Connection declined`, then failed to connect wallet
An error message also prints on console: `Unsupported wc_ method. wc_sessionPropose`

**SDK Version (if relevant)**
 - Client: JS
 - Version 2.4.5
 - @wagmi/core: 1.1.1
 - viem: 1.0.5

**To Reproduce**
1. Setup Web3Modal
```javascript
import { Web3Modal } from "@web3modal/html";
import { EthereumClient, w3mConnectors, w3mProvider } from '@web3modal/ethereum'
import { publicProvider } from "@wagmi/core/providers/public";
import { InjectedConnector } from '@wagmi/core/connectors/injected'
import { configureChains, createConfig, getWalletClient } from "@wagmi/core";
import { WalletConnectConnector } from '@wagmi/core/connectors/walletConnect';

const setupConnectModal = () => {
    const projectId = process.env.WALLET_CONNECT_PROJECT_ID

    const { chains, publicClient, webSocketPublicClient } = configureChains(
        [polygon],
        [w3mProvider({ projectId }), publicProvider()]
    );
    const wagmiConfig = createConfig({
        autoConnect: true,
        connectors: [
            new InjectedConnector({
                chains,
                options: {
                    shimDisconnect: true
                }
            }),
            new WalletConnectConnector({
                chains: chains,
                options: {
                    projectId: projectId,
                    isNewChainsStale: true,
                },
            }),
            ...w3mConnectors({ chains, version: 2, projectId })
        ],
        publicClient,
        webSocketPublicClient
    });

    const ethereumClient = new EthereumClient(wagmiConfig, chains);

    const web3Modal = new Web3Modal(
        { projectId },
        ethereumClient
    )

    return web3Modal
}

export const ConnectModal = setupConnectModal()
```

2. Open the modal
```javascript
import { ConnectModal } from "xxx";

...

await ConnectModal.openModal()
```

**Expected behavior**
Should connect wallet successfully.

**Screenshots**
Open the modal -> Scan QR code -> The modal shows `Connection declined`, then failed to connect wallet
An error message also prints on console: `Unsupported wc_ method. wc_sessionPropose`

Once I scan the QR code, the modal shows `Connection Request Declined`
![image](https://github.com/WalletConnect/walletconnect-monorepo/assets/132535663/d6ccebee-ec2d-429c-9441-9181ac12d2dc)

Error message is printed on console
<img width="624" alt="image" src="https://github.com/WalletConnect/walletconnect-monorepo/assets/132535663/625b8ca6-c19b-4283-8c38-9b86b3ba53c0">

The file points to `node_modules/@walletconnect/ethereum-provider/dist/index.es.js`

**Desktop (please complete the following information):**
 - OS: macOs 13.4
 - Browser Microsoft Edge
 - Version 114.0.1823.43

**Additional information**

I can connect to my dapp if I use WalletConnect v1 `WalletConnectLegacyConnector` (@wagmi/core/connectors/walletConnectLegacy)

**Comment by BigtoLenaDev** [comment]
Here's the [example](https://github.com/BigtoLenaDev/wagmi-mini-example), a simple run instruction is written in [README](https://github.com/BigtoLenaDev/wagmi-mini-example/blob/main/README.md).

The error message printed on console: 

![image](https://github.com/wagmi-dev/references/assets/132535663/0fda613e-7d1e-48f9-897c-c3dc1ea12d06)

**Comment by BigtoLenaDev** [comment]
Any update please?

**Comment by old-mikser** [comment]
same issue

**Comment by glitch-txs** [comment]
Hi, which wallets are you testing this with?

**Comment by BigtoLenaDev** [comment]
> Hi, which wallets are you testing this with?

I'm using InfinityWallet, if I use trust wallet, error message is `Chains must not be empty`

<img width="1127" alt="image" src="https://github.com/WalletConnect/walletconnect-monorepo/assets/132535663/e2fb4f64-85bd-4b76-9473-d0ab4fbc6a59">

**Comment by glitch-txs** [comment]
This issue has been fixed on the latest bersion of wagmi, please try npm i wagmi@latest viem@latest. I need to take a closer look to the other wallet, do you get the error above with InfinityWallet?
